torch/python/src/sphericart_torch/__init__.py

```diff
@@ -117,4 +125,70 @@ def compute(
             return self._sph.compute(xyz), None
 
 
-__all__ = ["SphericalHarmonics"]
+def e3nn_spherical_harmonics(
+    l_list: Union[List[int], int],
+    x: torch.Tensor,
+    normalize: Optional[bool] = False,
+    normalization: Optional[str] = "integral",
+) -> torch.Tensor:
+    """
+    Computes spherical harmonics with an interface similar to the e3nn package.
+
+    Provides an interface that is similar to :py:func:`e3nn.o3.spherical_harmonics`
+    but uses :py:class:`SphericalHarmonics` for the actual calculation.
+    Uses the same ordering of the [x,y,z] axes, and supports the same options for
+    input and harmonics normalization as :py:mod:`e3nn`. However, it does not support
+    defining the irreps through a :py:class:`e3nn.o3._irreps.Irreps` or a string
+    specification, but just as a single integer or a list of integers.
```

Comment:
How hard would it be to accept `Irreps` or string and turn them into a list of integers using code from e3nn?

Reply:
It's done in the spherical_harmonics implementation so we'd have to copy and paste a whole bunch of ifs, which I find a bit annoying.

```

        if isinstance(irreps_out, str):
            irreps_out = o3.Irreps(irreps_out)
        if isinstance(irreps_out, o3.Irreps) and irreps_in is None:
            for mul, (l, p) in irreps_out:
                if l % 2 == 1 and p == 1:
                    irreps_in = o3.Irreps("1e")
        if irreps_in is None:
            irreps_in = o3.Irreps("1o")

        irreps_in = o3.Irreps(irreps_in)
        if irreps_in not in (o3.Irreps("1x1o"), o3.Irreps("1x1e")):
            raise ValueError(f"irreps_in for SphericalHarmonics must be either a vector (`1x1o`) or a pseudovector (`1x1e`), not `{irreps_in}`")
        self.irreps_in = irreps_in
        input_p = irreps_in[0].ir.p  # pylint: disable=no-member

        if isinstance(irreps_out, o3.Irreps):
            ls = []
            for mul, (l, p) in irreps_out:
                if p != input_p**l:
                    raise ValueError(f"irreps_out `{irreps_out}` passed to SphericalHarmonics asked for an output of l = {l} with parity p = {p}, which is inconsistent with the input parity {input_p} — the output parity should have been p = {input_p**l}")
                ls.extend([l]*mul)
        elif isinstance(irreps_out, int):
            ls = [irreps_out]
        else:
            ls = list(irreps_out)
```

Reply:
If we want this to be a drop-in replacement, I think we need to copy-paste this code. We should also patch the `SphericalHarmonics` torch module. 

This way people can install sphericart_torch, run this function and everything is faster even if people are not using e3nn directly (e.g. they are trying to use allegro)